plugin/src/gatsby-node.ts

```diff
@@ -20,6 +20,7 @@ export function pluginOptionsSchema({ Joi }: PluginOptionsSchemaArgs) {
     storeUrl: Joi.string().required(),
     downloadImages: Joi.boolean(),
     verboseLogging: Joi.boolean(),
+    typeName: Joi.string().default(''),
```

Comment:
I wonder if we should have some validation on this property. I've noticed certain characters are not allowed. It seems that Joi does support Regex validation https://joi.dev/api/?v=17.4.0

Reply:
Do you know what characters are allowed? (or not allowed). If so I can provide the regex.

Reply:
I don't, but I noticed right away that slashes weren't allowed, for example:

```
Encountered an error parsing the provided GraphQL type definitions:
Syntax Error: Cannot parse the unexpected character "/".

  1 |
> 2 |     type foo/bar/baz/ShopifyProductVariant implements Node {
    |             ^
  3 |       product: foo/bar/baz/ShopifyProduct @link(from: "productId", by: "id")
  4 |       metafields: [foo/bar/baz/ShopifyMetafield] @link(from: "id", by: "productVariantId")
  5 |     }

```

So generally we could say that it's whatever the GraphQL type name rules are. But maybe for simplicity we could limit it to alphanumeric characters starting with an uppercase letter.

Reply:
Fixed! But, can we set a custom error message? Right now it'll throw and say `"typePrefix" with value "hemEU" fails to match the required pattern: /(^[A-Z][w|d]*)/`, (in this case it errors because it doesn't start with an uppercase letter) which isn't the most obvious explanation as to why it failed.

Reply:
Looks like it's supported, e.g.

```ts
    foo: Joi.string().pattern(/foo/).message("customMessage"),
```

Reply:
Perfect! You're better at reading the docs than me it seems 😅 . Fixed!